projects/social/social.py

```diff
@@ -22,51 +47,66 @@ def addFriendship(self, userID, friendID):
             self.friendships[userID].add(friendID)
             self.friendships[friendID].add(userID)
 
+# ---------------------------------------------------------------------------
+
     def addUser(self, name):
-        """
-        Create a new user with a sequential integer ID
-        """
-        self.lastID += 1  # automatically increment the ID to assign the new user
+        self.lastID += 1
         self.users[self.lastID] = User(name)
         self.friendships[self.lastID] = set()
 
-    def populateGraph(self, numUsers, avgFriendships):
-        """
-        Takes a number of users and an average number of friendships
-        as arguments
-
-        Creates that number of users and a randomly distributed friendships
-        between those users.
+# ---------------------------------------------------------------------------
 
-        The number of users must be greater than the average number of friendships.
-        """
-        # Reset graph
+    def populateGraph(self, numUsers, avgFriendships):
         self.lastID = 0
         self.users = {}
         self.friendships = {}
-        # !!!! IMPLEMENT ME
-
-        # Add users
+        for i in range(0, numUsers):
+            self.addUser(f"User {i}")
+        possibleFriendships = []
+        for userID in self.users:
+            for friendID in range(userID +1, self.lastID +1):
+                possibleFriendships.append((userID, friendID))
+        random.shuffle(possibleFriendships)
+        for i in range(0, (numUsers * avgFriendships // 2)):
+            friendship = possibleFriendships[i]
+            self.addFriendship(friendship[0], friendship[1])
 
-        # Create friendships
+# ---------------------------------------------------------------------------
 
     def getAllSocialPaths(self, userID):
-        """
-        Takes a user's userID as an argument
-
-        Returns a dictionary containing every user in that user's
-        extended network with the shortest friendship path between them.
-
-        The key is the friend's ID and the value is the path.
-        """
-        visited = {}  # Note that this is a dictionary, not a set
-        # !!!! IMPLEMENT ME
+        visited = {}
+        q = Queue()
+        q.enqueue([userID])
+        while q.size() > 0:
+            path = q.dequeue()
+            newUserID = path[-1]
+            if newUserID not in visited:
+                visited[newUserID] = path
+                for friendID in self.friendships[newUserID]:
+                    if friendID not in visited:
+                        new_path = list(path)
+                        new_path.append(friendID)
+                        q.enqueue(new_path)
         return visited
```

Comment:
The social paths and friendships graph look good to me!  Great work keeping the code readable and using meaningful variable names.